### PR TITLE
Refactorings/improvements after null-safety migration

### PR DESCRIPTION
**Pull request by algodave**
A few things need to be refactored/improved after null-safety migration, which was mainly performed by running the automatic migration tool.

### CODE REVIEW

File: lib/src/adapters/json_api.dart [hunk 1]
```diff
@@ -55,8 +55,9 @@ class JsonApiAdapter extends Adapter with Http {
     }
     JsonApiManyDocument cached = peekMany(endpoint, ids);
     if (cached.length != ids.length) {
-      List<JsonApiDocument?> cachedDocs = cached.toList();
-      Iterable<String?> cachedIds = cachedDocs.map((doc) => doc!.id);
+      List<JsonApiDocument> cachedDocs = cached.toList();
+      Iterable<String> cachedIds =
+          cachedDocs.map((doc) => doc.id).whereType<String>().toList();
```
Comment (Bringoff) [inline lib/src/adapters/json_api.dart:60]: @algodave whereType<String> is a nice alternative to cast, maybe we can make use of it in flutter-rest-data latest PR as well?

Comment (algodave) [reply lib/src/adapters/json_api.dart:60]: @Bringoff Yes, definitely

File: lib/src/adapters/json_api.dart [hunk 2]
```diff
@@ -220,9 +225,16 @@ class JsonApiAdapter extends Adapter with Http {
   }
 
   @override
-  JsonApiManyDocument peekMany(String endpoint, Iterable<String> ids) =>
-      JsonApiManyDocument(
-          ids.map((id) => peek(endpoint, id)).where((doc) => doc != null));
+  JsonApiManyDocument peekMany(String endpoint, Iterable<String> ids) {
```
Comment (algodave) [inline lib/src/adapters/json_api.dart:228]: @Bringoff This could be improved using `whereType()` or `whereNotNull()`

**Comment by algodave** [comment]
@Bringoff Please update Readme according to new things we introduced in this PR, so we can merge it. Thanks!

**Pull request merged**